ExportDialog.qml

```diff
@@ -0,0 +1,189 @@
+// Copyright (C) 2016 Allegorithmic
+//
+// This software may be modified and distributed under the terms
+// of the MIT license.  See the LICENSE file for details.
+
+import QtQuick 2.3
+import QtQuick.Window 2.2
+import QtQuick.Dialogs 1.2
+import QtQuick.Controls 1.4
+import QtQuick.Layouts 1.3
+
+import "photoshop.js" as Photoshop
+
+AlgDialog {
+    width: 400
+    height: 300
+    minimumWidth: 400
+    minimumHeight: 300
+    defaultButtonText: "Ok"
+    title: "Select Materials/Channels to export"
+
+    QtObject {
+        id: internal
+        property var documentType: {
+            'Material': "material",
+            'Stack': "stack",
+            'Channel': "channel"
+        }
+    }
+
+    function reload() {
+        var documentStructure = alg.mapexport.documentStructure()
+        documentStructureModel.clear()
+        var list = alg.settings.value("exportMaps", [])
+        var id = 0
+        for (var materialId in documentStructure.materials) {
+            var modelMaterialId = id
+            ++id
+            var material = documentStructure.materials[materialId]
+            documentStructureModel.append({'name': material.name,
+                                           'path': material.name,
+                                           'type': internal.documentType.Material,
+                                           'parentIndex': -1,
+                                           'defaultChecked': Photoshop.isMapExportable(list, material.name)})
+            //Browse stacks
+            for (var stackId in material.stacks) {
+                var stack = material.stacks[stackId]
+                var stackPath = material.name + "." + stack.name
+                var modelStackId = stack.name !== "" ? id : --id
+                ++id
+                // Do not display the main stack
+                if (stack.name !== "") {
+                    documentStructureModel.append({'name': stack.name,
+                                                      'path': stackPath,
+                                                      'type': internal.documentType.Stack,
+                                                      'parentIndex': modelMaterialId,
+                                                      'defaultChecked': Photoshop.isMapExportable(list, stackPath)})
+                }
+                //Browse channels
+                for (var channelId in stack.channels) {
```

Comment:
If there is only one channel in a stack (which has a name), we could merge as "stack/channel" label. It is mainly relevant when using custom material blending based on masks (as the fire hydrant sample).